### ISSUE TITLE
Feature request: animations with logarithmic speed

### ISSUE THREAD

**Issue by yo8192** [andymchugh/andrewbmchugh-flow-panel#98]
The speed of animations isn't "natural" for things which are more "exponential" in nature, for instance solar panel production or electricity consumption.

For instance, with settings like:
```
      thresholdOffValue: 10
      thresholdLwrValue: 100
      thresholdLwrDurationSecs: 1
      thresholdUprValue: 5000
      thresholdUprDurationSecs: 0.1
```

with the current formula (in Python) I get values like:
```
>>> [round(durationSecs(x, 100, 1, 5000, 0.1), 3) for x in [100, 200, 400, 800, 1600, 3200, 4700, 5000]] 
[1.0, 0.982, 0.945, 0.871, 0.724, 0.431, 0.155, 0.1]
```
The issue is that there is barely any difference between 200 and 400 even though it is twice as much.
Another way to look at it is that there is as much difference between 100 and 400 than between 4700 and 5000: 0.06. However 400 is 4 times 100 so we expect to see a big difference in speed, while 4700 is almost the same as 5000 so we expect little difference in speed.

I think a simple way to solve this would be to make the duration logarithmic, e.g. below https://github.com/andymchugh/andrewbmchugh-flow-panel/blob/4d72da2dc8d050c3ff908d3dc65f49aa052e2836/src/components/SvgUpdater.tsx#L341 add the typescript equivalent of:
```
    factor = math.log2(factor+1)
```

With this I get a spread of speeds which I think should feel more "natural":
```
>>> [round(durationSecsLog(x, 100, 1, 5000, 0.1), 3) for x in [100, 200, 400, 800, 1600, 3200, 4700, 5000]] 
[1.0, 0.974, 0.923, 0.827, 0.653, 0.364, 0.14, 0.1]
``` 

Would you consider giving this as a (configurable) option?

Thanks.

ps: if `log2` didn't feel right, possible alternatives that could be tried would be a `sin`-based function, e.g. in Python:
```
    factor = math.sin( (math.pi/2) * (v-thresholdLwrValue)/(thresholdUprValue-thresholdLwrValue))
```
or based on a circle formula, e.g. in Python:
```
    factor = (v-thresholdLwrValue)/(thresholdUprValue-thresholdLwrValue)
    factor = math.sqrt(2*factor-factor*factor)
```

**Comment by andymchugh** [comment]
Thanks for the time you've put into it. I'll have a think.

In the meantime you should be able to achieve this using the bespoke drive. With that you can use a mathjs formula on your data to create the dataRef that flows into the animFlow drive. I'd be interested to know how that goes. It's more boilerplate for you but it will give you complete control. At least it will unblock you for now.

**Comment by andymchugh** [comment]
I like your idea. I've implemented it as a power function so you get to specify the biasPower to shape the curve. It defaults to the current 1. Between values 0 and 1 you'll be ramping up early as you've asked for. Above 1 you'll be ramping up late. It will get built into the next version 1.16.4.

**Comment by andymchugh** [comment]
Feature available in 1.16.4, now published.

**Comment by yo8192** [comment]
Thanks @andymchugh , that's a nice way to implement it.
